geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -27,84 +27,75 @@ import org.locationtech.geomesa.accumulo.index.Strategy._
+import org.locationtech.geomesa.filter._
 import org.locationtech.geomesa.utils.geotools.RichAttributeDescriptors.RichAttributeDescriptor
 import org.locationtech.geomesa.utils.stats.IndexCoverage.IndexCoverage
 import org.locationtech.geomesa.utils.stats.{Cardinality, IndexCoverage}
-import org.opengis.feature.`type`.AttributeDescriptor
 import org.opengis.feature.simple.SimpleFeatureType
 import org.opengis.filter.expression.{Literal, PropertyName}
 import org.opengis.filter.temporal.{After, Before, During, TEquals}
 import org.opengis.filter.{Filter, PropertyIsEqualTo, PropertyIsLike, _}
 
-import scala.collection.JavaConversions._
 import scala.collection.JavaConverters._
 
-trait AttributeIdxStrategy extends Strategy with Logging {
+class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with Logging {
 
-  import org.locationtech.geomesa.accumulo.index.AttributeIndexStrategy._
+  import org.locationtech.geomesa.accumulo.index.AttributeIdxStrategy._
 
   /**
    * Perform scan against the Attribute Index Table and get an iterator returning records from the Record table
    */
-  def getAttributeIdxQueryPlan(query: Query,
-                               queryPlanner: QueryPlanner,
-                               attributeName: String,
-                               range: AccRange,
-                               output: ExplainerOutputType): QueryPlan = {
+  override def getQueryPlans(queryPlanner: QueryPlanner, hints: Hints, output: ExplainerOutputType) = {
+    val propsAndRanges = filter.primary.map(getPropertyAndRange(_, queryPlanner.sft))
+    val attributeName = propsAndRanges.head._1
+    val ranges = propsAndRanges.map(_._2)
+    // ensure we only have 1 prop we're working on
+    assert(propsAndRanges.forall(_._1 == attributeName))
 
     val sft = queryPlanner.sft
     val acc = queryPlanner.acc
-
-    output(s"Scanning attribute table for feature type ${sft.getTypeName}")
-    output(s"Range: ${ExplainerOutputType.toString(range)}")
-    output(s"Filter: ${query.getFilter}")
-
-    val attributeIterators = scala.collection.mutable.ArrayBuffer.empty[IteratorSetting]
-
-    val (geomFilters, otherFilters) = partitionGeom(query.getFilter, sft)
-    val (temporalFilters, nonSTFilters) = partitionTemporal(otherFilters, getDtgFieldName(sft))
-
-    output(s"Geometry filters: $geomFilters")
-    output(s"Temporal filters: $temporalFilters")
-    output(s"Other filters: $nonSTFilters")
-
-    val stFilter: Option[Filter] = filterListAsAnd(geomFilters ++ temporalFilters)
-    val ecqlFilter: Option[Filter] = filterListAsAnd(nonSTFilters)
-
     val encoding = queryPlanner.featureEncoding
     val version = acc.getGeomesaVersion(sft)
     val hasDupes = sft.getDescriptor(attributeName).isMultiValued
 
-    val kvsToFeatures = if (query.getHints.isBinQuery) {
+    val attributeIterators = scala.collection.mutable.ArrayBuffer.empty[IteratorSetting]
+
+    val (stFilter, ecqlFilter) = filter.secondary.map { f =>
```

Comment:
Happens in the join case - the attribute value in that case will only be the index value (e.g. date and geom). So we need to know if the filter can be satisfied by that, or if we need to apply it to the record table instead.